be/src/runtime/runtime_filter_worker.cpp

```diff
@@ -47,15 +47,29 @@ static inline std::shared_ptr<MemTracker> get_mem_tracker(const PUniqueId& query
     }
 }
 
-static void send_rpc_runtime_filter(doris::PBackendService_Stub* stub, RuntimeFilterRpcClosure* rpc_closure,
-                                    int timeout_ms, const PTransmitRuntimeFilterParams& request) {
+static void send_rpc_runtime_filter(TNetworkAddress dest, RuntimeFilterRpcClosure* rpc_closure, int timeout_ms,
+                                    int64_t http_min_size, const PTransmitRuntimeFilterParams& request) {
+    doris::PBackendService_Stub* stub = nullptr;
+    bool via_http = request.data().size() >= http_min_size;
+    if (via_http) {
+        if (auto res = BrpcStubCache::create_http_stub(dest); res.ok()) {
+            stub = res.value().release();
+        }
+    } else {
+        stub = ExecEnv::GetInstance()->brpc_stub_cache()->get_stub(dest);
+    }
+    if (stub == nullptr) {
+        LOG(WARNING) << strings::Substitute("The brpc stub of {}: {} is null.", dest.hostname, dest.port);
+        return;
+    }
+
     rpc_closure->ref();
     rpc_closure->cntl.Reset();
     rpc_closure->cntl.set_timeout_ms(timeout_ms);
-    // as the attachment is empty, the http rpc also can do like the following interface.
-    // create a http rpc stub: http_stub
-    // http_stub->transmit_runtime_filter(&rpc_closure->cntl, &request, &rpc_closure->result, rpc_closure);
     stub->transmit_runtime_filter(&rpc_closure->cntl, &request, &rpc_closure->result, rpc_closure);
+    if (via_http) {
+        delete stub;
```

Comment:
does  transmit_runtime_filter works in async mode?
if so,  stub seems be deleted before rpc finishes.

can we create a bundle of http stub beforehand for multiplexing?

Reply:
> does transmit_runtime_filter works in async mode? if so, stub seems be deleted before rpc finishes.
> 
> can we create a bundle of http stub beforehand for multiplexing?

yes, it works in async mode. According to brpc doc, we can use stub as a temporary variable and not necessary to keep it.
<img width="1016" alt="image" src="https://github.com/StarRocks/starrocks/assets/3675229/4a971bc5-cbea-436e-a2be-0a20402f6de1">